boards/arm/nrf9160_pca20035/CMakeLists.txt

```diff
@@ -4,12 +4,11 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 
-zephyr_library()
+if(CONFIG_TRUSTED_EXECUTION_SECURE AND NOT DEFINED CONFIG_MCUBOOT)
+	zephyr_library()
+	zephyr_library_sources(board_secure.c)
+endif()
 
 zephyr_library_sources_ifdef(
 	CONFIG_TRUSTED_EXECUTION_NONSECURE
 	board_nonsecure.c)
```

Comment:
You need to make sure there is no code path where this file is added to a library that hasn't been created.

Reply:
I'm not very familiar with cmake, should these source files be added some other way? Or is this the correct way to do it?

Reply:
Could we, instead, use the BOARD names, to compile-in these source files?

Reply:
@ioannisg Yes, that sounds like a better idea. As discussed elsewhere, `CONFIG_TRUSTED_EXECUTION_SECURE` will no longer be set in defconfig to align with PCA10090, so good idea to use board name instead to distinguish the two.

Reply:
@jtguggedal : I suspect that the CMake code is written incorrectly, but I can't be sure without knowing what kind of behaviour is desired. If you could express in english what you want to happen then I can translate that to CMake.

Reply:
@SebastianBoe I'll try: I want to compile in `board_secure.c` and `board_secure.c` depending on the Kconfig option. They're supposed to run during bootup, setting up the board control ICs, modem MAGPIO settings and such.

Reply:
Could they both be enabled simultaneously?

Could neither be enabled?

Reply:
`board_secure.c` should be run once every time the board boots, as it controls the onboard PMIC. `board_nonsecure.c` should only run for non-secure applications compiled for `nrf9160_pca20035ns` target